python/ray/dashboard/modules/serve/tests/test_serve_dashboard.py

```diff
@@ -358,12 +358,21 @@ def test_get_serve_instance_details(ray_start_stop, f_deployment_options, url):
             "docs_path": None,
             "deployments": {"f", "BasicDriver"},
             "source": "declarative",
+            "required_resources": {
+                "f": {
+                    "CPU": f_deployment_options.get("ray_actor_options", {}).get(
+                        "num_cpus", 0.1
+                    )
+                },
+                "BasicDriver": {"CPU": 0.1},
+            },
```

Comment:
don't know much about the serve internal pages, but is it weird that the internal schema is different from the external schema here?

i.e. required resources keyed by deployments at the app level vs required resources being a field on the deployment itself.

Reply:
Ah this is just an arbitrary way to store/retrieve the expected values for the test. You can see `test_schema.py::test_serve_instance_details_is_json_serializable` for an example of the actual format of the `required_resources`.